prowler/providers/aws/services/directoryservice/directoryservice_service.py

```diff
@@ -146,25 +153,38 @@ def __list_certificates__(self, regional_client):
                     directory.region == regional_client.region
                     and directory.type != DirectoryType.SimpleAD
                 ):
-                    list_certificates_paginator = regional_client.get_paginator(
-                        "list_certificates"
-                    )
-                    list_certificates_parameters = {"DirectoryId": directory.id}
-                    certificates = []
-                    for page in list_certificates_paginator.paginate(
-                        **list_certificates_parameters
-                    ):
-                        for certificate_info in page["CertificatesInfo"]:
-                            certificates.append(
-                                Certificate(
-                                    id=certificate_info["CertificateId"],
-                                    common_name=certificate_info["CommonName"],
-                                    state=certificate_info["State"],
-                                    expiry_date_time=certificate_info["ExpiryDateTime"],
-                                    type=certificate_info["Type"],
+                    try:
+                        list_certificates_paginator = regional_client.get_paginator(
+                            "list_certificates"
+                        )
+                        list_certificates_parameters = {"DirectoryId": directory.id}
+                        certificates = []
+                        for page in list_certificates_paginator.paginate(
+                            **list_certificates_parameters
+                        ):
+                            for certificate_info in page["CertificatesInfo"]:
+                                certificates.append(
+                                    Certificate(
+                                        id=certificate_info["CertificateId"],
+                                        common_name=certificate_info["CommonName"],
+                                        state=certificate_info["State"],
+                                        expiry_date_time=certificate_info[
+                                            "ExpiryDateTime"
+                                        ],
+                                        type=certificate_info["Type"],
+                                    )
                                 )
+                        self.directories[directory.id].certificates = certificates
+                    except ClientError as error:
+                        if (
+                            error.response["Error"]["Code"]
+                            == "UnsupportedOperationException"
+                        ):
+                            logger.warning(
+                                f"{directory.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
                             )
-                    self.directories[directory.id].certificates = certificates
+                            continue
```

Comment:
Is this `continue` right? Should it be one `tab` to the left?

Reply:
Changed